src/actions/index.js

```diff
@@ -3,11 +3,17 @@ import { FETCH_DATA } from './types';
 import Api from '../api/Api';
 
 export const actionFetchData = () => (dispatch) => {
+  dispatch({
+    type: FETCH_DATA,
```

Comment:
Create a new action type `FETCHING_DATA` and dispatch it with no payload (i.e. dispatch({ type: FETCHING_DATA }))

Better keep the shape of all actions to only have a type and a payload (& meta if required).